frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -121,39 +227,325 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+      className="co-alert"
+      title={t('olm~Successfully Uninstalled Operator')}
+      isInline
+    >
+      <p>
+        <Trans t={t} ns="olm">
+          Operator <strong>{{ name }}</strong> successfully uninstalled from{' '}
+          <strong>{{ namespace }}</strong>.
+        </Trans>
+      </p>
+    </Alert>
+  );
+};
+
+const OperatorUninstallError: React.FC<{ operatorUninstallErrorMessage: string }> = ({
+  operatorUninstallErrorMessage,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <Alert
+      variant="danger"
+      className="co-alert"
+      title={t('olm~Error Uninstalling Operator')}
+      isInline
+    >
+      <p>
+        {t('olm~There was an error uninstalling the operator.')}
+        <br />
+        {operatorUninstallErrorMessage}
+      </p>
+    </Alert>
+  );
+};
+
+const OperandDeletionErrors: React.FC<{
+  operandDeletionErrors: OperandWithErrorProps[];
+  csvName: string;
+  cancel?: () => void;
+}> = ({ operandDeletionErrors, csvName, cancel }) => {
+  const { t } = useTranslation();
+  return (
+    <Alert variant="danger" className="co-alert" title={t('olm~Error Deleting Operands')} isInline>
+      <p>
+        <Trans t={t} ns="olm">
+          There were errors deleting the following Operands, they will need to be deleted manually:
+        </Trans>
+      </p>
+      <OperandErrors
+        operandsWithErrors={operandDeletionErrors}
+        csvName={csvName}
+        cancel={cancel} // for breadcrumbs & cancel modal when clicking on operand links
+      />
+    </Alert>
+  );
+};
+
+const OperandDeletionSuccess: React.FC<{ name: string; namespace: string }> = ({
+  name,
+  namespace,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <Alert
+      variant="success"
+      className="co-alert"
+      title={t('olm~Successfully deleted all Operand Instances')}
+      isInline
+    >
+      <p>
+        <Trans t={t} ns="olm">
+          All Operand instances for Operator <strong>{{ name }}</strong> in{' '}
+          <strong>{{ namespace }}</strong> have been deleted.
+        </Trans>
+      </p>
+    </Alert>
+  );
+};
+
+const SubmitResults: React.FC<SubmitResultsProps> = ({
+  operatorUninstallErrorMessage,
+  name,
+  namespace,
+  isDeleteAllOperandsSelected,
+  operandDeletionErrors,
+  csvName,
+  cancel,
+}) => {
+  const alerts = [];
+  alerts.push(
+    operatorUninstallErrorMessage ? (
+      <OperatorUninstallError operatorUninstallErrorMessage={operatorUninstallErrorMessage} />
+    ) : (
+      <OperatorUninstallSuccess name={name} namespace={namespace} />
+    ),
+  );
+  if (isDeleteAllOperandsSelected) {
+    alerts.push(
+      areOperandDeletionErrors(operandDeletionErrors) ? (
+        <OperandDeletionErrors
+          operandDeletionErrors={operandDeletionErrors}
+          csvName={csvName}
+          cancel={cancel}
+        />
+      ) : (
+        <OperandDeletionSuccess name={name} namespace={namespace} />
+      ),
+    );
+  }
+  return <>{alerts}</>;
```

Comment:
Could we break this down a bit more so that we don't have the unusual array logic?

```suggestion
const UninstallAlert = ({ error, name, namespace }) =>
  error ? (
    <OperatorUninstallError error={error} />
  ) : (
    <OperatorUninstallSuccess name={name} namespace={namespace} />
  );

const OperandDeletionAlert = (errors, csvName, cancel, name, namespace) =>
  errors.length ? (
    <OperandDeletionErrors operandDeletionErrors={errors} csvName={csvName} cancel={cancel} />
  ) : (
    <OperandDeletionSuccess name={name} namespace={namespace} />
  );

const SubmitResults: React.FC<SubmitResultsProps> = ({
  operatorUninstallErrorMessage,
  name,
  namespace,
  isDeleteAllOperandsSelected,
  operandDeletionErrors,
  csvName,
  cancel,
}) => (
  <>
    <UninstallAlert error={operatorUninstallErrorMessage} name={name} namespace={namespace} />
    {isDeleteAllOperandsSelected && (
      <OperandDeletionAlert
        cancel={cancel}
        csvName={csvName}
        errors={operandDeletionErrors}
        name={name}
        namespace={namespace}
      />
    )}
  </>
);
```